### ISSUE TITLE
Cannot delete corrupt image

### ISSUE THREAD

**Issue by pdaigle** [vmware/vic#6513]
#For bug reports, please include the information below:

**VIC version:**

vSphere Integrated Containers v1.2.1-13858-c3db65f

**Deployment details:**

./vic-machine-linux create \
--target vcsa-01a.corp.local \
--user administrator@corp.local \
--compute-resource RegionA01-COMP01 \
--image-store RegionA01-ISCSI01-COMP01 \
--volume-store RegionA01-ISCSI01-COMP01:default \
--public-network VM-RegionA01-vDS-COMP \
--public-network-ip 192.168.100.22/24 \
--public-network-gateway 192.168.100.1 \
--dns-server 192.168.110.10 \
--container-network VM-RegionA01-vDS-COMP:routable \
--container-network-firewall VM-RegionA01-vDS-COMP:open \
--bridge-network Bridge01-RegionA01-vDS-COMP \
--name virtual-container-host-1.2.1 \
--tls-cname=vch1.corp.local \
--registry-ca=/etc/docker/certs.d/registry.corp.local/ca.crt \
--whitelist-registry registry.corp.local \
--whitelist-registry registry.hub.docker.com \

**Steps to reproduce:**

Interrupted an image being downloaded/extracted to the image cache before it can complete.

**Actual behavior:**

I am unable to run this image (because it is incomplete/corrupted) and unable to delete the image from the image cache.

![image](https://user-images.githubusercontent.com/8586715/31205606-4e94ce6c-a927-11e7-9049-27e87430900b.png)

**Logs:**

[logs.zip](https://github.com/vmware/vic/files/1357702/logs.zip)

**Comment by mlh78750** [comment]
Needs attention. What is the implication of the failure. Can you redownload and recover? Is there a work around to remove.

@mdubya66 I want this triage and understood before it's deferred from 1.3.

**Comment by gigawhitlocks** [comment]
Been trying to reproduce this behavior all day with no luck.
I tried these approaches:
1) Cancelling during download (many times)
2) Cancelling during extraction (many times)
3) Deleting ImageMetadata files directly in the datastore browser
4) Deleting random layers after pull, directly in the datastore browser 
5) Deleting bits of the attached volume for dch-photon, directly in the datastore browser

After any of the above, I was able to remove the image as long as it was reported in `docker images`. For 1) and 2), sometimes the image wouldn't show up in `docker images` because not enough of the image was downloaded, but I was able to resume `pull` without issues.

FWIW the "Invalid configuration for device (1)" error message is seen in this issue we've had open for a long time https://github.com/vmware/vic/issues/4666 and I suspect this issue is related to that.
